examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/llm/run_clm_no_trainer.py

```diff
@@ -0,0 +1,451 @@
+
+    if args.peft_model_id is not None:
+        from peft import PeftModel
+        user_model = PeftModel.from_pretrained(user_model, args.peft_model_id)
+
+    # to channels last
+    user_model = user_model.to(memory_format=torch.channels_last)
+    user_model.eval()
+    return user_model, tokenizer
+
+if args.quantize:
+    # dataset
+    user_model, tokenizer = get_user_model()
+    calib_dataset = load_dataset(args.dataset, split="train")
+    # calib_dataset = datasets.load_from_disk('/your/local/dataset/pile-10k/') # use this if trouble with connecting to HF
+    calib_dataset = calib_dataset.shuffle(seed=args.seed)
+    calib_evaluator = Evaluator(calib_dataset, tokenizer, args.batch_size, pad_max=args.pad_max_length, is_calib=True)
+    calib_dataloader = DataLoader(
+        calib_evaluator.dataset,
+        batch_size=calib_size,
+        shuffle=False,
+        collate_fn=calib_evaluator.collate_batch,
+    )
+
+    def calib_func(prepared_model):
+        for i, calib_input in enumerate(calib_dataloader):
+            if i > args.calib_iters:
+                break
+            prepared_model(calib_input[0])
+
+    recipes = {}
+    eval_func = None
+    from neural_compressor import PostTrainingQuantConfig, quantization
+    # specify the op_type_dict and op_name_dict
+
+    if args.approach == 'weight_only':
+        op_type_dict = {
+            '.*':{ 	# re.match
+                "weight": {
+                    'bits': args.woq_bits, # 1-8 bits 
+                    'group_size': args.woq_group_size,  # -1 (per-channel)
+                    'scheme': args.woq_scheme, # sym/asym
+                    'algorithm': args.woq_algo, # RTN/AWQ/TEQ
+                },
+            },
+        }
+        op_name_dict={
+            'lm_head':{"weight": {'dtype': 'fp32'},},
+            'embed_out':{"weight": {'dtype': 'fp32'},},  # for dolly_v2
+        }
+        recipes["rtn_args"] = {
+            "enable_mse_search": args.woq_enable_mse_search,
+            "enable_full_range": args.woq_enable_full_range,
+        }
+        recipes['gptq_args'] = {
+                'percdamp': args.gptq_percdamp, 
+                'act_order':args.gptq_actorder, 
+                'block_size': args.gptq_block_size, 
+                'nsamples': args.gptq_nsamples, 
+                'use_max_length': args.gptq_use_max_length,
+                'pad_max_length': args.gptq_pad_max_length
+            }
+        # GPTQ: use assistive functions to modify calib_dataloader and calib_func
+        # TEQ: set calib_func=None, use default training func as calib_func
+        if args.woq_algo in ["GPTQ", "TEQ"]:
+            calib_func = None
+
+        conf = PostTrainingQuantConfig(
+            approach=args.approach,
+            op_type_dict=op_type_dict,
+            op_name_dict=op_name_dict,
+            recipes=recipes,
+        )
+
+        # for test on various models, keep the code of directly call gptq_quantize
+        if args.gptq_debug:
```

Comment:
This branch has some issues, you can edit the code based on https://github.com/intel/neural-compressor/pull/1475/files#diff-6bf78d064d7b7861d36fc73e88a0e1c4c75c21b44bc598abbbdd4cc508a70981R294